src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -1489,13 +1577,22 @@ public static String verifyVirtualMachineImage(
             final ServicePrincipal servicePrincipal,
             final String location,
             final String storageAccountName,
+            final String imageTopLevelType,
             final AzureVMAgentTemplate.ImageReferenceType referenceType,
+            final String buildInImage,
             final String image,
             final String imagePublisher,
             final String imageOffer,
             final String imageSku,
             final String imageVersion) {
-        if ((referenceType == AzureVMAgentTemplate.ImageReferenceType.UNKNOWN && StringUtils.isNotBlank(image))
+        if (imageTopLevelType == null || imageTopLevelType.equals("base")) {
```

Comment:
`isTopLevelType` is bind on a `AzureVMAgentTemplate `instance, to compare given type equals the type inside template or not, so it seems we can't call `isTopLevelType` here.